[16315] Implement Community Channel option: view channel members and details

ibrkhalil opened the pull request:
fixes #16315

### Summary

Implements community channel option: view details and members screen.
![CleanShot 2024-01-23 at 14 19 06](https://github.com/status-im/status-mobile/assets/33176106/4300d8ce-1d1e-4a9e-8ba1-5169a68c1f16)


status: ready <!-- Can be ready or wip -->

ibrkhalil committed:
Implement Community Channel option: view channel members and details

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 8521f828 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-18489/1/) | 2024-01-14 19:42:13 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-18489/1/consoleText) |

ibrkhalil committed:
Add notes

ibrkhalil committed:
Progress

ibrkhalil committed:
improvements

ibrkhalil committed:
Improvements

ilmotta commented on src/quo/components/text_combinations/view.cljs:
```diff
@@ -7,16 +7,30 @@
+               :margin-top  1}}
+      source]]
+    [rn/image
+     {:source (if (string? source)
```
This check is unnecessary because `rn/image` already does this. cc @flexsurfer since I saw a comment of yours in a PR asking why we do this in `rn/image`.

ilmotta commented on src/quo/components/text_combinations/view.cljs:
```diff
@@ -7,16 +7,30 @@
+  [source size emoji? emoji-background-color]
+  (if emoji?
+    [rn/view
+     {:style {:border-radius    50
```
Should be moved to `style` namespace.

ilmotta commented on src/quo/components/text_combinations/view.cljs:
```diff
@@ -28,15 +42,20 @@
      (when avatar
        [rn/view {:style style/avatar-container}
-        [icon avatar 32]])
+        [icon avatar 32 nil nil]])
```
You can just make `icon` take a single map and then you won't need to pass nils.

ilmotta commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,113 @@
+  (let [{:keys [description chat-name emoji muted
+                chat-type]
+         :as   chat}  (rf/sub [:chats/chat-by-id
+                               chat-id])
```
`chat-id` can be moved up to the previous line.

ilmotta commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,113 @@
+         :title                           [quo/channel-name
+                                           {:channel-name chat-name
+                                            :unlocked?    true}]
+         :emoji                           (when (seq emoji) emoji)
```
`emoji` is a string. Using `seq` on a string is not very precise nor the fastest because Clojure has to transform the string to an IndexedSeq. Use `(not (string/blank? emoji))`.

See results on the JVM, 40x faster. Longer strings give the same results.

```clojure
  (criterium/with-progress-reporting
    (criterium/bench
        (seq "Lorem Ipsum is simply")))
  ;; Evaluation count : 1442390700 in 60 samples of 24039845 calls.
  ;; Execution time mean : 38.201201 ns
  ;; Execution time std-deviation : 0.678546 ns
  ;; Execution time lower quantile : 36.352233 ns ( 2.5%)
  ;; Execution time upper quantile : 39.278308 ns (97.5%)
  ;; Overhead used : 5.455626 ns

  (criterium/with-progress-reporting
    (criterium/bench
        (not (string/blank? "Lorem Ipsum is simply"))))
  ;; Evaluation count : 10148449380 in 60 samples of 169140823 calls.
  ;; Execution time mean : 0.914537 ns
  ;; Execution time std-deviation : 0.071185 ns
  ;; Execution time lower quantile : 0.681369 ns ( 2.5%)
  ;; Execution time upper quantile : 0.984959 ns (97.5%)
  ;; Overhead used : 5.455626 ns
```

And by precise I mean the developer doesn't have any hint as to what type `emoji` is. I had to lookup the implementation of `text-combinations` to see that it's in fact a string.

ilmotta commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,113 @@
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn contact-item-render
```
Functions can be made private in this namespace.

ilmotta commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,113 @@
+    :render-fn                         contact-item-render
+    :scroll-event-throttle             8}])
+
+(defn f-view
```
Why is this component functional?

ibrkhalil replied:
Fixed

ilmotta commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,113 @@
+         :as   chat}  (rf/sub [:chats/chat-by-id
+                               chat-id])
+        pins-count    (rf/sub [:chats/pin-messages-count chat-id])
+        items         (rf/sub [:communities/sorted-community-members-section-list
```
You are adding subscriptions outside the renderer, which makes the component non-reactive. Most certainly a bug to the user. If anything this component needs gets updated in the app db, it won't re-render. Saving very specific use cases, subscriptions should be called inside the renderer.

ibrkhalil replied:
Apparently, I forgot to push the last commit :)
Also I had fixed the text for the muted/unmuted status there, Here it was hardcoded to mute channel

ilmotta commented on src/status_im/subs/communities.cljs:
```diff
@@ -67,6 +61,41 @@
           (sort-by #(visibility-status-utils/visibility-status-order (get % 0)))))))
 
+(re-frame/reg-sub
+ :communities/sorted-community-members-section-list
```
This sub could really get some refactors. This is not status-go where we functions have hundreds of lines of code and that's okay for everybody ;)

Jokes aside, more importantly, I don't see unit tests for it and it should have plenty given the amount of logic. This is in our guidelines too.

ibrkhalil replied:
Added a test, Will try to refactor it.

ibrkhalil replied:
Removed the call to reverse as it was reordering the elements in linear time, And instead made the initial sorting of elements be done in reverse order.
Will test it more though

ibrkhalil committed:
Refactors

ibrkhalil committed:
Add subscription test

ilmotta commented on src/status_im/subs/communities_test.cljs:
```diff
@@ -503,3 +504,52 @@
+        :visibility-status-updates
+        {"0x01" {:status-type 3}
+         "0x02" {:status-type 3}})
+      (is (= '({:title :Offline
```
We don't need to quote the list. `=` will succeed comparing to a vector too.

ibrkhalil replied:
I wanted to make `clj-kondo` happy
![CleanShot 2024-01-31 at 12 31 22](https://github.com/status-im/status-mobile/assets/33176106/1f19eb20-4bcf-4dcc-bd6b-cfd60a2240a5)
And `make lint` failed
```clj
src/status_im/subs/communities_test.cljs:552:23: error: a string is not a function
src/status_im/subs/communities_test.cljs:554:23: error: a string is not a function
linting took 28129ms, errors: 2, warnings: 1218
make: *** [lint] Error 3
ibrkhalil@ibrkhalil status-mobile % 
```

ilmotta replied:
@ibrkhalil, a list can't be used as a data structure without being quoted, otherwise Clojure will try to use the first argument as a function. That's why you see the complaint.

But in tests, we use vectors for equality checks because most of the time we don't actually care if it's a vector instance or a lazy seq for example.

ibrkhalil replied:
Got it, Thanks Icaro

ilmotta commented on src/status_im/subs/communities.cljs:
```diff
@@ -67,6 +61,41 @@
+                         {}
+                         (keys members))]
+     (->> members
+          (sort-by #(get names (get % 0)))
```
The logic to get sorted communities already exists in sub `:communities/sorted-community-members`. Instead of copying it, please extract into a common function and reuse.

ibrkhalil replied:
Done

ilmotta commented on src/status_im/subs/communities_test.cljs:
```diff
@@ -503,3 +504,52 @@
+      (swap! rf-db/app-db assoc :profile/profile profile-test/sample-profile)
+      (swap! rf-db/app-db assoc
+        :visibility-status-updates
+        {"0x01" {:status-type 3}
```
3 can be replaced by the var in `constants` namespace to improve readability.

ibrkhalil replied:
Done

ilmotta commented on src/status_im/subs/communities.cljs:
```diff
@@ -67,6 +61,41 @@
+                                                  my-status-update
+                                                  (get visibility-status-updates public-key))
+                         visibility-status-type (:status-type status-update)]
+                     (or (= visibility-status-type constants/visibility-status-automatic)
```
This final logic to check if it's online is duplicated from the sub `:visibility-status-updates/online?`

ilmotta commented on src/status_im/subs/communities.cljs:
```diff
@@ -67,6 +61,41 @@
+                         {}
+                         (keys members))]
+     (->> members
+          (sort-by #(get names (get % 0)))
```
`(get % 0)` -> `(first %)`

ilmotta commented on src/status_im/subs/communities.cljs:
```diff
@@ -67,6 +61,41 @@
+          reverse
+          (map (fn [[k v]]
+                 {:title (if k :Online :Offline)
+                  :data  (flatten v)}))))))
```
Why do flatten `v`? Isn't it already just a one-dimensional sequence of public keys?

ibrkhalil replied:
On my initial implementation the value was nested collections, Fixed.

ilmotta commented on src/status_im/subs/communities.cljs:
```diff
@@ -67,6 +61,41 @@
+     (->> members
+          (sort-by #(get names (get % 0)))
+          keys
+          (group-by (fn [item]
```
Can be just `(group-by online?)`

ilmotta commented on src/status_im/subs/communities.cljs:
```diff
@@ -67,6 +61,41 @@
+                         visibility-status-type (:status-type status-update)]
+                     (or (= visibility-status-type constants/visibility-status-automatic)
+                         (= visibility-status-type constants/visibility-status-always-online))))
+         names   (reduce (fn [acc contact-identity]
```
This whole `names` function is duplicated from the sub above `:communities/sorted-community-members`.

ilmotta commented on src/status_im/subs/communities.cljs:
```diff
@@ -67,6 +61,41 @@
+                      (online? item)))
+          reverse
+          (map (fn [[k v]]
+                 {:title (if k :Online :Offline)
```
Any reason to use keywords instead of strings as the titles? Normally keywords in clojure are only lowercase.

ibrkhalil committed:
Refactors

ibrkhalil committed:
Refactors

ilmotta commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,108 @@
+    :sticky-section-headers-enabled    false
+    :render-section-header-fn          contact-list/contacts-section-header
+    :render-section-footer-fn          footer
+    :render-fn                         contact-item-render
```
Prefer a noun and not use the word `render`, so just `contact-item`. I get the `:render-fn` name doesn't help with the confusion 🤷🏼

ibrkhalil committed:
Use vectors in test

ibrkhalil committed:
Really use vectors :)

ibrkhalil committed:
Update test

ibrkhalil committed:
Lint

flexsurfer commented on translations/en.json:
```diff
@@ -2479,5 +2479,7 @@
     "apply-changes": "Apply changes",
-    "receiver-networks-warning": "Changing these settings may result in sending tokens to networks the recipient doesn't use"
+    "receiver-networks-warning": "Changing these settings may result in sending tokens to networks the recipient doesn't use",
+    "online-community-member": "Online",
```
we already have two! "Online", i would say we should have only one

flexsurfer replied:
so only `"online": "Online"` should be used

ibrkhalil replied:
Understood, Addressed.

flexsurfer commented on src/quo/components/text_combinations/style.cljs:
```diff
@@ -15,3 +15,14 @@
+
+(defn textual-emoji
+  [size emoji-background-color]
+  {:style {:border-radius    50
```
styles functions should return only map of styles, not map of props

flexsurfer commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,108 @@
+  [rn/section-list
+   {:key-fn                            :public-key
+    :content-container-style           {:padding-bottom 20}
+    :get-item-layout                   (fn [_ index]
```
better to do not use anon functions here

flexsurfer commented on src/status_im/subs/communities.cljs:
```diff
@@ -5,6 +5,7 @@
     [legacy.status-im.ui.screens.profile.visibility-status.utils :as visibility-status-utils]
     [re-frame.core :as re-frame]
     [status-im.constants :as constants]
+    [status-im.subs.general :as general-subs]
```
better to have a separate namespace for subs.utils

ibrkhalil replied:
Can you take another look please?

ibrkhalil committed:
Refactors

J-Son89 commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,108 @@
+  (:require [clojure.string :as string]
+            [quo.core :as quo]
+            [quo.foundations.colors :as colors]
+            [quo.theme :as quo.theme]
```
nit `[quo.theme]`

ibrkhalil committed:
Remove some options when pressing on self on members options

ibrkhalil committed:
Lint

ibrkhalil committed:
Fix translations

status-im-auto commented:
## 96% of end-end tests have passed
```
Total executed tests: 48
Failed tests: 1
Expected to fail tests: 1
Passed tests: 46
```
```
IDs of failed tests: 702732 
```
```
IDs of expected to fail tests: 703503 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=18489&TR_CASE_IDS=702732">Rerun failed tests</a></li><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486974">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//*[@content-desc='pinned-messages-menu']//*[starts-with(@text,'Message 4')]/../../*[@content-desc='pinned-by']/android.widget.TextView` </div><div>Device 2: `Text` is `user admin` </div></blockquote></p>

```
critical/chats/test_group_chat.py:404: in test_group_chat_pin_messages
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Pinned messages count 4 doesn't match expected 3 for user 2
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/401004462a974d3a94a9860f70ff925d?auth=509244e1a9684c06db9965bf100bac75#345">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6e3f50cdc34f403ba8bab6feebf928a4?auth=8b4586ec1c65156ca91429b95fe541b4#246">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/20869c3aaecc4c15ae94de724748f97f?auth=882c48f3ebdff4ab49058d21a4664c76#241">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487016">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Curated communities not loading, https://github.com/status-im/status-mobile/issues/17852]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (46)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487009">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#212">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#167">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486999">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#241">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#231">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486990">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#291">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#287">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487007">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#553">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#432">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487001">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#397">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#353">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487019">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#592">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#462">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486989">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#195">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#138">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486988">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#153">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#109">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486993">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#315">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#302">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486992">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#171">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#126">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486991">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a04c7c15a8942eaa42ff56061949b16?auth=eb2b8425eeb96a54510232d4621217e3#360">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8d25f705d91429fb61802a7b706da39?auth=b95d108f0247f14e903a287ad443fd1d#343">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486982">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f3578df09e414255a4369dfaf81e7493?auth=b8e1a609226b77216f50693e1dae11da#173">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66809365e8a74a15b04833c84f3d755b?auth=c326b5951d10e12b50e6adb48ed15cc0#128">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486981">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f3578df09e414255a4369dfaf81e7493?auth=b8e1a609226b77216f50693e1dae11da#86">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66809365e8a74a15b04833c84f3d755b?auth=c326b5951d10e12b50e6adb48ed15cc0#87">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487015">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f3578df09e414255a4369dfaf81e7493?auth=b8e1a609226b77216f50693e1dae11da#101">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66809365e8a74a15b04833c84f3d755b?auth=c326b5951d10e12b50e6adb48ed15cc0#115">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487018">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ee989d55b09e484ba06c79cc2bd7a93a?auth=d38c1e77943f61ec1537ff315302321c#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486978">test_links_deep_links</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ee989d55b09e484ba06c79cc2bd7a93a?auth=d38c1e77943f61ec1537ff315302321c#199">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486986">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e70224b91b9844c6808a56a897f10262?auth=7318e2a39ea79962b0d01ec3199d24c4#190">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/67e8cb51015e4c86af599e286c53ec81?auth=ee9a9aa4b04f40f4df4aa33c33ec8e43#177">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487004">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e70224b91b9844c6808a56a897f10262?auth=7318e2a39ea79962b0d01ec3199d24c4#331">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/67e8cb51015e4c86af599e286c53ec81?auth=ee9a9aa4b04f40f4df4aa33c33ec8e43#332">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486983">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e70224b91b9844c6808a56a897f10262?auth=7318e2a39ea79962b0d01ec3199d24c4#134">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/67e8cb51015e4c86af599e286c53ec81?auth=ee9a9aa4b04f40f4df4aa33c33ec8e43#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486994">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e70224b91b9844c6808a56a897f10262?auth=7318e2a39ea79962b0d01ec3199d24c4#328">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/67e8cb51015e4c86af599e286c53ec81?auth=ee9a9aa4b04f40f4df4aa33c33ec8e43#301">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487017">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e70224b91b9844c6808a56a897f10262?auth=7318e2a39ea79962b0d01ec3199d24c4#544">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/67e8cb51015e4c86af599e286c53ec81?auth=ee9a9aa4b04f40f4df4aa33c33ec8e43#521">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487005">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f49b24a29214e6fb25df60deb18970d?auth=4a62eaa6b2f082b8129c811d4c96e169#119">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4eed42545681474396dcf7007dd44194?auth=ca414b38140b20ec1ddb122ce54dec57#103">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487006">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f49b24a29214e6fb25df60deb18970d?auth=4a62eaa6b2f082b8129c811d4c96e169#167">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4eed42545681474396dcf7007dd44194?auth=ca414b38140b20ec1ddb122ce54dec57#119">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487008">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e1d125710d3e4c1f9c07905886478d8d?auth=b43a103ccd905abed34e9777604a9945#252">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486976">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e1d125710d3e4c1f9c07905886478d8d?auth=b43a103ccd905abed34e9777604a9945#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487000">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e1d125710d3e4c1f9c07905886478d8d?auth=b43a103ccd905abed34e9777604a9945#108">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486995">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e1d125710d3e4c1f9c07905886478d8d?auth=b43a103ccd905abed34e9777604a9945#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487012">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e1d125710d3e4c1f9c07905886478d8d?auth=b43a103ccd905abed34e9777604a9945#127">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487014">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/401004462a974d3a94a9860f70ff925d?auth=509244e1a9684c06db9965bf100bac75#473">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6e3f50cdc34f403ba8bab6feebf928a4?auth=8b4586ec1c65156ca91429b95fe541b4#292">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/20869c3aaecc4c15ae94de724748f97f?auth=882c48f3ebdff4ab49058d21a4664c76#259">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487011">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/401004462a974d3a94a9860f70ff925d?auth=509244e1a9684c06db9965bf100bac75#264">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6e3f50cdc34f403ba8bab6feebf928a4?auth=8b4586ec1c65156ca91429b95fe541b4#179">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/20869c3aaecc4c15ae94de724748f97f?auth=882c48f3ebdff4ab49058d21a4664c76#148">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487010">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/401004462a974d3a94a9860f70ff925d?auth=509244e1a9684c06db9965bf100bac75#156">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6e3f50cdc34f403ba8bab6feebf928a4?auth=8b4586ec1c65156ca91429b95fe541b4#109">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/20869c3aaecc4c15ae94de724748f97f?auth=882c48f3ebdff4ab49058d21a4664c76#89">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486984">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/401004462a974d3a94a9860f70ff925d?auth=509244e1a9684c06db9965bf100bac75#137">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6e3f50cdc34f403ba8bab6feebf928a4?auth=8b4586ec1c65156ca91429b95fe541b4#81">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/20869c3aaecc4c15ae94de724748f97f?auth=882c48f3ebdff4ab49058d21a4664c76#81">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486985">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/401004462a974d3a94a9860f70ff925d?auth=509244e1a9684c06db9965bf100bac75#308">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6e3f50cdc34f403ba8bab6feebf928a4?auth=8b4586ec1c65156ca91429b95fe541b4#220">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/20869c3aaecc4c15ae94de724748f97f?auth=882c48f3ebdff4ab49058d21a4664c76#207">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487002">test_navigation_jump_to</a>, id: 702936 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44f94f682de4de98e61cc9744e74fa7?auth=b85e1ff265bb7b80d714f9db2277181f#139">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cf1054fa911b4de5b21fdd0f5f7c7e7d?auth=4bda4bbca2f11428beaabb3cc510823a#122">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487003">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44f94f682de4de98e61cc9744e74fa7?auth=b85e1ff265bb7b80d714f9db2277181f#167">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cf1054fa911b4de5b21fdd0f5f7c7e7d?auth=4bda4bbca2f11428beaabb3cc510823a#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486980">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8532f2e5ac3f40b1b71f6b194255cf37?auth=37681ef0931a4aa96cccc8cca439b23e#142">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f863c81bf2748cba2159632c24fa335?auth=1cdb241c20ace3a12166c26df68506b8#122">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486975">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8532f2e5ac3f40b1b71f6b194255cf37?auth=37681ef0931a4aa96cccc8cca439b23e#554">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f863c81bf2748cba2159632c24fa335?auth=1cdb241c20ace3a12166c26df68506b8#397">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486987">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8532f2e5ac3f40b1b71f6b194255cf37?auth=37681ef0931a4aa96cccc8cca439b23e#408">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f863c81bf2748cba2159632c24fa335?auth=1cdb241c20ace3a12166c26df68506b8#259">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486977">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8532f2e5ac3f40b1b71f6b194255cf37?auth=37681ef0931a4aa96cccc8cca439b23e#350">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f863c81bf2748cba2159632c24fa335?auth=1cdb241c20ace3a12166c26df68506b8#207">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486998">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8532f2e5ac3f40b1b71f6b194255cf37?auth=37681ef0931a4aa96cccc8cca439b23e#463">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f863c81bf2748cba2159632c24fa335?auth=1cdb241c20ace3a12166c26df68506b8#294">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3487013">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8532f2e5ac3f40b1b71f6b194255cf37?auth=37681ef0931a4aa96cccc8cca439b23e#491">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f863c81bf2748cba2159632c24fa335?auth=1cdb241c20ace3a12166c26df68506b8#337">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486973">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8532f2e5ac3f40b1b71f6b194255cf37?auth=37681ef0931a4aa96cccc8cca439b23e#204">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f863c81bf2748cba2159632c24fa335?auth=1cdb241c20ace3a12166c26df68506b8#167">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486972">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8532f2e5ac3f40b1b71f6b194255cf37?auth=37681ef0931a4aa96cccc8cca439b23e#86">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f863c81bf2748cba2159632c24fa335?auth=1cdb241c20ace3a12166c26df68506b8#87">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486979">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7808f82936e4571ae5965baf8f234a2?auth=aec1f6a89b5239ac42a27333e0251e51#141">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4a5c31f4f0c542deb51c475e96706721?auth=1a6555718a07b4a1abb60d29a5df68c2#184">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486997">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7808f82936e4571ae5965baf8f234a2?auth=aec1f6a89b5239ac42a27333e0251e51#101">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4a5c31f4f0c542deb51c475e96706721?auth=1a6555718a07b4a1abb60d29a5df68c2#104">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3486996">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7808f82936e4571ae5965baf8f234a2?auth=aec1f6a89b5239ac42a27333e0251e51#59">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4a5c31f4f0c542deb51c475e96706721?auth=1a6555718a07b4a1abb60d29a5df68c2#62">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

mariia-skrypnyk commented:
Hi @ibrkhalil !
Please resolve conflicts and rebase your PR.
Thanks!

ibrkhalil committed:
clean

ibrkhalil commented:
> Hi @ibrkhalil ! Please resolve conflicts and rebase your PR. Thanks!

Done

J-Son89 commented on src/quo/components/text_combinations/view.cljs:
```diff
@@ -29,7 +33,9 @@
-           emoji-hash]
+           emoji-hash
+           emoji
+           emoji-background-color]
```
@ibrkhalil - here the principle is similar to the other or component, we don't want to add override colors. In this I believe this is just `customization-color` which we pass as that for consistency among the quo components. You then have to resolve that accordingly

J-Son89 commented on src/quo/components/text_combinations/style.cljs:
```diff
@@ -16,3 +16,14 @@
+   :height           size
+   :justify-content  :center
+   :align-items      :center
+   :background-color emoji-background-color})
```
<img width="427" alt="Screenshot 2024-03-26 at 00 45 25" src="https://github.com/status-im/status-mobile/assets/22799766/66125df5-fca8-4e1f-91de-ef40e43a0dae">

If that is related to this, then this is 

`(colors/resolve-color customization-color theme 10)`

J-Son89 commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,108 @@
+                                           {:channel-name chat-name
+                                            :unlocked?    true}]
+         :emoji                           (when (not (string/blank? emoji)) emoji)
+         :emoji-background-color          (colors/resolve-color profile-color theme 10)
```
please do this in the component and just pass `customization-color`. We want the design system to encapsulate styles so we can configure it by props :)

J-Son89 replied:
Also it's nice if our quo components have a consistent api so we know what to expect.

J-Son89 commented on src/status_im/contexts/communities/actions/channel_view_details/view.cljs:
```diff
@@ -0,0 +1,108 @@
+                                    (home.actions/mute-chat-action chat-id chat-type muted)))}]}]]]
+       [members items profile-color]])))
+
+(defn view [] (quo.theme/with-theme view-internal))
```
since this is new code it's better to not use the `with-theme` and instead make use of `quo.theme/use-theme-value` hook directly in the view 👍

J-Son89 commented on src/status_im/navigation/screens.cljs:
```diff
@@ -140,6 +142,10 @@
      :options   {:insets {:top? true}}
      :component communities.accounts-selection/view}
 
+    {:name      :view-channel-members-and-details
```
consider adding new screens with :screen/<context>.<keyword>
e.g `:screen/chat.view-channel-members-and-details`

see https://github.com/status-im/status-mobile/pull/18778

J-Son89 commented on src/status_im/subs/communities.cljs:
```diff
@@ -48,12 +48,22 @@
- :<- [:communities]
- (fn [[{:keys [community-id]} communities]]
-   (get-in communities [community-id :members])))
+(defn keys->names
```
private?

J-Son89 commented on src/status_im/subs/communities.cljs:
```diff
@@ -48,12 +48,22 @@
+          {}
+          public-keys))
+
+(defn sort-members-by-name
```
private?

J-Son89 commented on src/status_im/subs/utils.cljs:
```diff
@@ -0,0 +1,7 @@
+(ns status-im.subs.utils
```
let's avoid very generic namespaces like this :)

This will eventually be filled with completely unrelated functionality. 

At the very least `status-im.subs.chat.utils`

J-Son89 commented on src/status_im/subs/general.cljs:
```diff
@@ -6,7 +6,7 @@
     [legacy.status-im.utils.mobile-sync :as mobile-network-utils]
     [re-frame.core :as re-frame]
-    [status-im.constants :as constants]
+    [status-im.subs.utils :as utils]
```
also consider requiring as `chat.utils` or `chat-utils`. utils is very generic

mariia-skrypnyk commented:
**ISSUE 3: Incorrect color of channel on Channels details screen**

**Steps:**

1. Login to app
2. Open any community
3. Look at the colours chosen for these channels
4. Long tap on any channel 
5. Go to View channel members and details

**Actual result:** incorrect colour of chosen channel
<img width="239" alt="Screenshot 2024-03-26 at 11 23 31" src="https://github.com/status-im/status-mobile/assets/147824399/f320b381-dfae-4ca7-8638-3b0fb1119252">
<img width="323" alt="Screenshot 2024-03-26 at 11 24 32" src="https://github.com/status-im/status-mobile/assets/147824399/25025efe-becf-4f22-89fd-2b1aa76eeb68">

Expected result: Color should be correct

mariia-skrypnyk commented:
Hi @ibrkhalil @codemaster115  !

Any update on it?
If no - just ping me here when it's time to test! 🙌

codemaster115 commented:
> Hi @ibrkhalil @codemaster115 !
> 
> Any update on it? If no - just ping me here when it's time to test! 🙌

@mariia-skrypnyk I will update it soon

ibrkhalil committed:
clean

ibrkhalil committed:
clean

ibrkhalil committed:
clean

ibrkhalil committed:
cleanup

ibrkhalil commented:
@J-Son89 Thank you so much Jamie for re-reviewing, Sorry I had a lot of other tasks in my queue and couldn't address your comments at the time.

I've addressed them now.

ibrkhalil committed:
Wrapup

ibrkhalil committed:
Finalize

ibrkhalil commented:
@mariia-skrypnyk
Hey Mariia, I have addressed issues 1 and 2.
But for issue 3 I am a little bit confused, Should the channel avatar in view-details screen have the profile customization color or the community customization color, or the channel customization color/
Here
![image](https://github.com/status-im/status-mobile/assets/33176106/7708d412-1c0e-446c-bc93-07e2b9401e7b)


CC: @Francesca-G

ibrkhalil committed:
cleanup

Francesca-G commented:
> @mariia-skrypnyk Hey Mariia, I have addressed issues 1 and 2. But for issue 3 I am a little bit confused, Should the channel avatar in view-details screen have the profile customization color or the community customization color, or the channel customization color
> 
> CC: @Francesca-G

It should have channel customization color 👍

<img width="338" alt="Screenshot 2024-04-03 alle 10 21 47" src="https://github.com/status-im/status-mobile/assets/122543089/82540e4d-dbf9-4926-906d-a93be9d0dd85">

ibrkhalil committed:
Colorfy

ibrkhalil committed:
Lint

mariia-skrypnyk commented:
Great! @ibrkhalil 

Can you please rebase and resolve conflicts?

ibrkhalil commented:
> Great! @ibrkhalil
> 
> Can you please rebase and resolve conflicts?

Done

ibrkhalil committed:
clean

ibrkhalil commented:
> @ibrkhalil Great!
> 
> Issue 3 is resolved and tested! Maybe we can create followups for @codemaster115 to continue working not in this PR? I can do it as do not want to pause merge for so long.

Actually Mariia, I added fixes for all 3 issues.
Let's check on develop after merging if the other two still persist and if so let's create follow ups.

ibrkhalil committed:
Merge branch 'develop' into 16315

mariia-skrypnyk commented:
Hi @ibrkhalil !

✅ Issue 1 
Great job!

🟡 Issue 2: pinned messages are visible only when they are new ones (pinned during testing) but not for pinned messages in the past.
Can you fix it?

ibrkhalil commented:
> Hi @ibrkhalil !
> 
> 
> 
> ✅ Issue 1 
> 
> Great job!
> 
> 
> 
> 🟡 Issue 2: pinned messages are visible only when they are new ones (pinned during testing) but not for pinned messages in the past.
> 
> 
> 
> **Steps:**
> 
> Open channel details with previously pinned message in the past (not today)
> 
> **Result:** zero pinned messages on Channel Details screen
> 
> 
> 
> Can you fix it? 

Yes, Taking a look

mariia-skrypnyk commented:
> Yes, Taking a look

Seems like it is our known unresolved fetching problem.
Not related to your PR. 
Thanks! Great job! 💪
Move it to @Francesca-G design review! 🔬

ibrkhalil committed:
Merge branch 'develop' into 16315

Francesca-G reviewed (commented):
[Here](https://www.figma.com/file/18CoEYIaS6nbFxEpW1FX1M/Review-Communities-%2F-Mobile?type=design&node-id=70%3A34248&mode=design&t=ZfL7j53ScifmUAWe-1) is the design review :)

ibrkhalil commented:
> [Here](https://www.figma.com/file/18CoEYIaS6nbFxEpW1FX1M/Review-Communities-%2F-Mobile?type=design&node-id=70%3A34248&mode=design&t=ZfL7j53ScifmUAWe-1) is the design review :)

Centered the icon, Added the compressed keys for contacts and added jump-to button, Added the gradient
The missing profile pictures is another issue and is currently being worked on by someone else and is unrelated.

ibrkhalil committed:
Merge branch 'develop' into 16315

ibrkhalil committed:
Design review comments

ibrkhalil committed:
Merge branch 'develop' into 16315
